scaffolding-chef/lib/scaffolding.ps1

```diff
@@ -0,0 +1,133 @@
+#
+# A scaffolding for Chef Policyfile packages
+#
+
+if (!$scaffold_policy_name) {
+  Write-Host "You must set `$scaffold_policy_name to a valid policy name. For example:`n `$scaffold_policy_name=base `n Will build a base.rb policyfile."
+  exit 1
+}
+
+function Load-Scaffolding {
+  $scaffold_chef_client="stuartpreston/chef-client-detox"
+  $scaffold_chef_dk="core/chef-dk"
+
+  $pkg_deps+=@("${scaffold_chef_client}", "core/cacerts")
+  $pkg_build_deps+=@("${scaffold_chef_dk}", "core/git")
+  $pkg_svc_run="set_just_so_you_will_render"
+}
+
+function Invoke-DefaultBuildService {
+  New-Item -ItemType directory -Path "${pkg_prefix}/hooks"
+
+Add-Content -Path "$pkg_prefix/hooks/run" -Value @"
+function Chef-Client {
+  {{pkgPathFor "stuartpreston/chef-client-detox"}}/bin/chef-client.bat -z -l {{cfg.log_level}} -c $pkg_svc_config_path/client-config.rb -j $pkg_svc_config_path/attributes.json --once --no-fork --run-lock-timeout {{cfg.run_lock_timeout}}
+}
+
+`$splay_duration = Get-Random -InputObject (0..{{cfg.splay}}) -Count 1
+
+`$splay_first_run_duration = Get-Random -InputObject (0..{{cfg.splay_first_run}}) -Count 1
+
+`$env:SSL_CERT_FILE="{{pkgPathFor "core/cacerts"}}/ssl/cert.pem"
+
+cd {{pkg.path}}
+
+Start-Sleep -Seconds `$splay_first_run_duration
+Chef-Client
+
+while(`$true){
+  Start-Sleep -Seconds `$splay_duration
+  Start-Sleep -Seconds {{cfg.interval}}
+  Chef-Client
+}
+"@
+}
+
+
+function Invoke-DefaultBuild {
+  if(-Not (Test-Path -Path "${scaffold_policyfile_path}")){
+    Write-BuildLine "Could not detect a policyfiles directory, this is required to proceed!"
+    exit 1
```

Comment:
This code works 100% fine, I'm not sure what the objection is here?

The goal here is that if the policyfile_path is defined to something that doesn't exist that it will stop the build.

Reply:
Okay yep I've had a poor experience with using `exit` in PowerShell but if it functions I"m good with it.